lib/iris/coords.py

```diff
@@ -1345,7 +1345,9 @@ def __add__(self, mod):
         return Cell(point, bound)
 
     def __hash__(self):
-        return super().__hash__()
+        if self.bound is None:
+            return hash(self.point)
+        return hash((self.point, *sorted(self.bound)))
```

Comment:
I'm not too sure about the `sorted` here: I tried to mimic what the equals operator does, but sorting may only work if bounds is just two points.

Reply:
I got rid of the sorted, I think it should work now.